File: ios/Sources/VaultSync/Domain/Repositories/CryptoService.swift
import Foundation

public protocol CryptoService: Sendable {
    func encrypt(data: Data, keyAlias: String) async throws -> EncryptedPayload
    func decrypt(payload: EncryptedPayload, keyAlias: String) async throws -> Data
    func calculateSha256(data: Data) async -> String
}

public extension CryptoService {
    var defaultKeyAlias: String { "vaultsync_master_key" }
}

File: android/app/src/main/java/com/vaultsync/domain/repository/CryptoService.kt
package com.vaultsync.domain.repository

import com.vaultsync.domain.model.EncryptedPayload

interface CryptoService {
    suspend fun encrypt(data: ByteArray, keyAlias: String = DEFAULT_KEY_ALIAS): EncryptedPayload
    suspend fun decrypt(payload: EncryptedPayload, keyAlias: String = DEFAULT_KEY_ALIAS): ByteArray
    suspend fun calculateSha256(data: ByteArray): String

    companion object {
        const val DEFAULT_KEY_ALIAS = "vaultsync_master_key"
    }
}
